signbank/dictionary/adminviews.py

```diff
@@ -249,6 +252,69 @@ def subquery_render_to_csv_response(self, context):
 
         return response
 
+    def ready_for_validation_render_to_csv_response(self, context):
+        if not self.request.user.has_perm("dictionary.export_csv"):
+            msg = _("You do not have permissions to export to CSV.")
+            messages.error(self.request, msg)
+            raise PermissionDenied(msg)
+
+        # Create the HttpResponse object with the appropriate CSV header.
+        # It is a Python file-like object.
+        response = HttpResponse(content_type="text/csv; charset=utf-8")
+        response[
+            "Content-Disposition"
+        ] = 'attachment; filename="ready-for-validation-export.csv"'
+
+        # Set up for outputting CSV.
+        writer = csv.writer(response)
+
+        # The queryset may or may not already be filtered for the ready for validation tag.
+        # We have to make sure it is filtered by the tag, so we are filtering again
+        ready_for_validation_qs = TaggedItem.objects.get_union_by_model(
+            self.get_queryset(),
+            Tag.objects.filter(name=settings.TAG_READY_FOR_VALIDATION)
+        )
+
+        csv_queryset = (
+            ready_for_validation_qs
+            .prefetch_related("glossvideo_set", "glosstranslations_set")
+            .annotate(
+                gloss_main_aggregate=StringAgg(
+                    GlossTranslations.objects.filter(
+                        gloss=OuterRef("pk"),
+                        language__language_code_2char="EN"
+                    ).values("translations")[:1],
+                    self.CSV_AGG_DELIM,
+                    distinct=True,
+                    output_field=CharField()
+                ),
+            )
+        )
+
+        headers = [
+            "idgloss",
+            "gloss_main",
+            "video_url"
+        ]
+        writer.writerow(headers)
+
+        for gloss_record in csv_queryset:
+            row = [gloss_record.idgloss, gloss_record.gloss_main_aggregate]
+            validation_video = gloss_record.glossvideo_set.filter(
```

Comment:
**suggestion**: this code will not use the prefetch from above as a filter has been applied.

instead of doing `filter().first()` on every loop iteration, consider either:
1. creating a lookup dict before the loop with the list of gloss record primary keys
  i.e.: the existing filter + `gloss_id__in=ids`
2. retrieving the entire glossvideo_set and determining whether to include it using python code